File: serene/src/Serene.Web/Modules/ServerTypes/Workflow/WorkflowDefinitionColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { WorkflowDefinitionRow } from "./WorkflowDefinitionRow";

export interface WorkflowDefinitionColumns {
    WorkflowKey: Column<WorkflowDefinitionRow>;
    Name: Column<WorkflowDefinitionRow>;
    InitialState: Column<WorkflowDefinitionRow>;
}

export class WorkflowDefinitionColumns extends ColumnsBase<WorkflowDefinitionRow> {
    static readonly columnsKey = 'Workflow.WorkflowDefinition';
    static readonly Fields = fieldsProxy<WorkflowDefinitionColumns>();
}

File: serene/src/Serene.Web/Modules/WorkflowManagement/WorkflowDefinitionColumns.cs
namespace Serene.WorkflowManagement.Forms;

[ColumnsScript("Workflow.WorkflowDefinition")]
[BasedOnRow(typeof(Serenity.Workflow.Entities.WorkflowDefinitionRow), CheckNames = true)]
public class WorkflowDefinitionColumns
{
    [EditLink]
    public string WorkflowKey { get; set; }
    public string Name { get; set; }
    public string InitialState { get; set; }
}
